clients/admin-ui/src/features/datastore-connections/datastore-connection.slice.ts

```diff
@@ -5,22 +5,61 @@ import type { RootState } from "../../app/store";
 import { BASE_URL, CONNECTION_ROUTE } from "../../constants";
 import { selectToken } from "../auth";
 import {
+  ConnectionType,
   DatastoreConnection,
   DatastoreConnectionParams,
   DatastoreConnectionResponse,
   DatastoreConnectionStatus,
+  DisabledStatus,
+  SystemType,
+  TestingStatus,
 } from "./types";
 
+// function appendToQueryString(queryString:string, )
+
 function mapFiltersToSearchParams({
   search,
   page,
   size,
-}: Partial<DatastoreConnectionParams>): any {
-  return {
-    ...(search ? { search } : {}),
-    ...(page ? { page: `${page}` } : {}),
-    ...(typeof size !== "undefined" ? { size: `${size}` } : {}),
-  };
+  connection_type,
+  test_status,
+  system_type,
+  disabled_status,
+}: Partial<DatastoreConnectionParams>): string {
+  let queryString = "";
```

Comment:
In the future, could be refactored to push each query prop / val to an array of maps, then concatenate with `&` at the end. This would avoid all the checks for `queryString`.